### PR TITLE
Prometheus: Various buffered and streaming parsing fixes

### PR DESCRIPTION
**Pull request by toddtreece**
**What this PR does / why we need it**:
 
- I attempted to copy most of the utility functions were copied directly from the `buffered` code, but apparently I made a slight modification to the logic in the legend formatting code, which resulted in `__auto` not being handled properly.
- Added some additional tests, and pointed buffered & streaming tests at the same golden files so that we can ensure that the output matches. This was not possible before because `buffered` responses were using `*float64` value columns with `nil` values, but it seems as buffered changed to `float64` at some point recently, so now the output should match.
- Buffered responses needed to stop skipping `NaN` to match the streaming behavior

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by itsmylife — APPROVED** [review]
❤️ 🚀

**Review by ryantxu — APPROVED** [review]

File: pkg/tsdb/prometheus/buffered/time_series_query.go [hunk 1]
```diff
@@ -379,11 +379,7 @@ func matrixToDataFrames(matrix model.Matrix, query *PrometheusQuery, frames data
 
 		for i, k := range v.Values {
 			timeField.Set(i, k.Timestamp.Time().UTC())
-			value := float64(k.Value)
-
-			if !math.IsNaN(value) {
-				valueField.Set(i, value)
-			}
+			valueField.Set(i, float64(k.Value))
```
Comment (toddtreece) [inline pkg/tsdb/prometheus/buffered/time_series_query.go:382]: @kylebrandt @itsmylife in order to bring buffered in line with the streaming response, i needed to stop skipping `NaN` in buffered responses. is there an issue here with alerting?

Comment (itsmylife) [reply pkg/tsdb/prometheus/buffered/time_series_query.go:382]: @kylebrandt is the right person for this question.

**Review by itsmylife — APPROVED** [review]

**Review by kylebrandt — APPROVED** [review]
Thank you!

**Pull request merged**